spec/models/order_import_spec.rb

```diff
@@ -1,342 +1,376 @@
-    opts[:product_name]       || "Example Item",
-    opts[:quantity]           || 1,
-    opts[:order_date]         || nucore_format_date(default_order_date),
-    opts[:fulfillment_date]   || nucore_format_date(default_fulfilled_date),
-    opts[:note]               || "Test note"
-  ])
-
-  @order_import.errors_for(row)
-end
+require "spec_helper"
+require "controller_spec_helper"
+
+require "stringio"
+require "csv_helper"
+
+CSV_HEADERS = [
+  "Netid / Email",
+  "Chart String",
+  "Product Name",
+  "Quantity",
+  "Order Date",
+  "Fulfillment Date",
+  "Note",
+]
 
 def nucore_format_date(date)
   date.strftime("%m/%d/%Y")
 end
 
 describe OrderImport do
+  subject(:order_import) do
+    OrderImport.create!(
+      created_by: director.id,
+      upload_file: stored_file,
+      facility: facility,
+    )
+  end
+
+  let(:account_users_attributes) do
+    account_users_attributes_hash(user: guest) +
+    account_users_attributes_hash(
+      user: guest2,
+      created_by: guest,
+      user_role: AccountUser::ACCOUNT_PURCHASER,
+    )
+  end
+  let(:csv_row) { CSVHelper::CSV::Row.new(CSV_HEADERS, row_data) }
   let(:default_order_date) { 4.days.ago.to_date }
   let(:default_fulfilled_date) { 3.days.ago.to_date }
+  let(:director) { @director }
+  let(:error_file_row_count) do
+    Paperclip
+      .io_adapters
+      .for(order_import.error_file.file)
+      .read
+      .split("\n")
+      .count
+  end
+  let(:facility) { create(:facility) }
+  let(:facility_account) do
+    facility.facility_accounts.create!(attributes_for(:facility_account))
+  end
   let(:fiscal_year_beginning) { SettingsHelper::fiscal_year_beginning }
-
-  before(:all) do
-    create_users
+  let(:guest) { @guest }
+  let(:guest2) { create(:user, username: "guest2") }
+  let(:import_errors) { order_import.import_row(csv_row).errors }
+  let(:import_file_row_count) { import_file.read.split("\n").count }
+  let(:item) do
+    facility.items.create!(attributes_for(:item,
+      facility_account_id: facility_account.id,
+      name: "Example Item",
+    ))
+  end
+  let(:service) do
+    facility.services.create!(attributes_for(:service,
+      facility_account_id: facility_account.id,
+      name: "Example Service",
+    ))
   end
+  let(:stored_file) do
+    StoredFile.create!(
+      file: StringIO.new("c,s,v"),
+      file_type: "import_upload",
+      name: "clean_import.csv",
+      created_by: director.id,
+    )
+  end
+
+  let(:row_data) {[ username, account_number, product_name, quantity, order_date, fulfillment_date, note ]}
+  let(:username) { guest.username }
+  let(:account_number) { "111-2222222-33333333-01" }
+  let(:product_name) { "Example Item" }
+  let(:quantity) { 1 }
+  let(:order_date) { nucore_format_date(default_order_date) }
+  let(:fulfillment_date) { nucore_format_date(default_fulfilled_date) }
+  let(:note) { "Test note" }
+
+  before(:all) { create_users }
 
   before :each do
     Timecop.freeze(fiscal_year_beginning + 5.days)
```

Comment:
There's a `:timecop_freeze` metatag option you could use, and then not have to worry about the `Timecop.return` later on. You just need to add a `let(:now)` for defining what time to freeze at.